### DISCUSSION TITLE
Blocking or allowing specific clients not working

### DISCUSSION THREAD

**Discussion by oshimapt** [AdguardTeam/AdGuardHome#5649] [General]
Hello,
I'm using adguard home version v0.107.26 on a opnsense box.
The adguard is working great if I use global settings, but not working for specific clients.
Ex. If I create a persistent client, identify the client with is IP address and try to block/allow Youtube service, it won't work.
If I block/allow Youtube globally it works fine.

This can be verified with custom filters. If I create a filter ||example.org^ it will block the host without a problem and the check if a host is filtered shows the blocked host, but if I use ||example.com^$client='myphone' it won't work and the check if a host is filtered shows no results.

I understand that if a rule is not made correctly it will be ignored, but I can't see to find what I'm doing wrong.

**Comment by oshimapt** [comment]
hello,

just found that if i use this example from the wiki (||example.org^$ctag=device_pc|device_phone) it wont work either...
it looks like some modifiers are not working the it is supposed

**Comment by ainar-g** [comment]
I can reproduce neither of these issues.  Both forms work for me when I add them to the *Custom filtering rules*.  Please make sure that:

1.  You use the client **name** as opposed the ClientID when using the `client` modifier.

2.  You do not use quotation marks with the `ctag` identifier.

**Reply by oshimapt** [reply]
Hello,

Narrowed down to mobile phones and smart TV, all running Android.

The issue is the same, if I block a service globally it works, but for individual client it won't work.
The clients are identified by the IP address and it is fixed.

**Reply by ainar-g** [reply]
Make sure you put AdGuard Home's address into both DNS address inputs.  Otherwise, Android will default to using Google DNS.

**Reply by oshimapt** [reply]
i've reconfigured the DHCP to give the second dns as you told.
but i cant test it because i'm not at home right now

**Reply by joeyfigaro** [reply]
> i've reconfigured the DHCP to give the second dns as you told. but i cant test it because i'm not at home right now

Did it work?
